coreos/kdump: Add kdump e2e test using mco

gursewak1997 opened the pull request:
Add e2e test for OCP CI that validates enabling kdump and generating kernel core via machine config successfully. This is one of the steps to to enhance the kdump feature.

gursewak1997 commented:
/retest-required

gursewak1997 commented:
> ```Standard error from the command:
> failed: all tests must define a [sig-XXXX] tag or have a rule "[Top Level] kdump TestKdump"
> exit status 1
> ```

Yup I am going over the [doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests) to add the relevant tags before I re-commit.

gursewak1997 commented:
/retest

gursewak1997 commented:
/retest-required

gursewak1997 commented:
/retest-required

gursewak1997 commented:
/retest

miabbott commented on test/extended/coreos/kdump.go:
```diff
@@ -0,0 +1,143 @@
+	runCommandAndRetry(createKdumpMC)
+	runCommandAndRetry(createInfraPool)
+
+	// Create infra pool to roll out MC changes
```
I'm not familiar with the broader origin tests, so this might be obvious, but...why do we have to create a infra pool here?  Can't we just apply the MC to the worker pool?

miabbott replied:
Oh I think I get it...the infra pool is just to select a single node to run the test on rather than all the worker nodes.

miabbott commented:
Another question for due to unfamiliarity...when/where does this new test run?  Will it be part of CI jobs?

I'm struggling to find evidence that this test was run in any of the CI jobs that ran against the PR.  (Furthermore, I can only see the other `[sig-coreos]` test running as part of `e2e-gcp`

miabbott commented:
/test help

cgwalters commented:
See e.g. https://github.com/openshift/machine-config-operator/commit/825be33519852121fc1cc94695d1a759fb7e218b which we need to copy into privileged pods now as part of a recent security policy change

gursewak1997 commented:
/test e2e-aws-serial

gursewak1997 commented:
/retest

gursewak1997 commented:
/retest

stbenjam commented:
/test

stbenjam commented:
We've been bitten hard by some unreliable serial tests lately, I'd feel better if I saw it pass a couple times on some different configurations.

/test e2e-aws-serial
/test e2e-gcp-fips-serial
/test e2e-gcp-fips-serial
/test e2e-metal-ipi-serial
/test e2e-metal-ipi-serial-ovn-ipv6
/test e2e-openstack-serial

travier commented:
Note that this test actively makes a node crash, so I'm not sure how we should account for that in general.

stbenjam commented:
> Note that this test actively makes a node crash, so I'm not sure how we should account for that in general.

So far on the one aws run, it didn't seem to be a problem.  We have some synthetic tests that hunts for segfaults and go panics, but I don't think anything is looking for kernel panics.  Does the node recover? I'm wondering if it might not because machine-config goes degraded....

See this run: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27291/pull-ci-openshift-origin-master-e2e-aws-serial/1550551517881176064

```
{Operator degraded (RequiredPoolsFailed): Failed to resync 4.12.0-0.ci.test-2022-07-22-185749-ci-op-59ryclwt-latest because: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool worker is not ready, retrying. Status: (pool degraded: true total: 3, ready 2, updated: 2, unavailable: 1)]  Operator degraded (RequiredPoolsFailed): Failed to resync 4.12.0-0.ci.test-2022-07-22-185749-ci-op-59ryclwt-latest because: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool worker is not ready, retrying. Status: (pool degraded: true total: 3, ready 2, updated: 2, unavailable: 1)]}
```

travier commented:
As it's a fake crash, the node should just reboot after dumping the crash dump. The unavailability should be temporary.

travier commented:
> See this run: [prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27291/pull-ci-openshift-origin-master-e2e-aws-serial/1550551517881176064](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27291/pull-ci-openshift-origin-master-e2e-aws-serial/1550551517881176064)
> 
> ```
> {Operator degraded (RequiredPoolsFailed): Failed to resync 4.12.0-0.ci.test-2022-07-22-185749-ci-op-59ryclwt-latest because: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool worker is not ready, retrying. Status: (pool degraded: true total: 3, ready 2, updated: 2, unavailable: 1)]  Operator degraded (RequiredPoolsFailed): Failed to resync 4.12.0-0.ci.test-2022-07-22-185749-ci-op-59ryclwt-latest because: error during syncRequiredMachineConfigPools: [timed out waiting for the condition, error pool worker is not ready, retrying. Status: (pool degraded: true total: 3, ready 2, updated: 2, unavailable: 1)]}
> ```

But indeed this would be the symptoms if the node would not reboot for any reason.

gursewak1997 commented:
/test e2e-aws-disruptive

gursewak1997 commented:
I updated the tags for this test to `[Slow]`(Since the test typically took more than 5 minutes to finish), and `[Disruptive]`(Since it includes rebooting a node). Also, since any `[Disruptive]` test is also assumed to qualify for the `[Serial]` label, but need not be labelled as both as per this [doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests), I dropped the `[Serial]` label.
I am not too sure in which job the kdump test should run now because I don't see it in ci/prow/e2e-aws-disruptive` or `ci/prow/e2e-aws-serial`
On the other hand, I have also updated the test not to have any degraded nodes after the test finishes.
